File: radar-commons-server/src/main/java/org/radarcns/config/ServerConfigMixin.java
package org.radarcns.config;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;

/** Mixin class to load ServerConfig files with. */
public abstract class ServerConfigMixin {
    /** Proxy host name. Null if not set. */
    @JsonProperty("proxy_host")
    public abstract String getProxyHost();

    @JsonProperty("proxy_port")
    public abstract String getProxyPort();

    @JsonSetter("path")
    public abstract void setPath(String path);
}

File: radar-commons-server/src/main/java/org/radarbase/config/ServerConfigMixin.kt
package org.radarbase.config

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonSetter

/** Mixin class to load ServerConfig files with.  */
abstract class ServerConfigMixin {
    @get:JsonProperty("proxy_host")
    abstract val proxyHost: String?

    @get:JsonProperty("proxy_port")
    abstract val proxyPort: String?

    @JsonSetter("path")
    abstract fun setPath(path: String?)
}
